editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationController.java

```diff
@@ -0,0 +1,310 @@
+          }
+        });
+      }
+      // if the selected feature is a point, select the feature
+      else if (selectedFeature.getGeometry().getGeometryType() == GeometryType.POINT) {
+        ((FeatureLayer) layerResult.getLayerContent()).selectFeature(selectedFeature);
+        // open a dialog to edit the feature's attributes
+        showEditableAttributes(selectedFeature);
+      } else {
+        new Alert(Alert.AlertType.WARNING, "Feature of unexpected geometry type selected.").show();
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a dialog with text fields to allow editing of the given feature's 'AD_ADDRESS' and
+   * 'ST_STR_NAM' attributes.
+   */
+  private void showEditableAttributes(Feature selectedFeature) {
+
+    // create a dialog to edit the attributes of the selected feature
+    EditAttributesDialog editAttributesDialog = new EditAttributesDialog(selectedFeature);
+
+    // show the dialog and wait for the user response
+    editAttributesDialog.showAndWait();
+  }
+
+  /**
+   * Update the attributes of the selected feature.
+   */
+  private void updateAttributes(Feature selectedFeature) {
+
+    // update feature in the feature table
+    ListenableFuture<Void> editResult = selectedFeature.getFeatureTable().updateFeatureAsync(selectedFeature);
+    editResult.addDoneListener(() -> {
+              try {
+                if (editResult.isDone()) {
+                  editResult.get();
+                }
+              } catch (InterruptedException | ExecutionException e) {
+                new Alert(Alert.AlertType.ERROR, "Cannot update attributes: " + e.getCause().getMessage()).show();
+              }
+            }
+    );
+  }
+
+  /**
+   * Move the selected point feature to the given map point by updating the selected
+   * feature's geometry and feature table.
+   */
+  private void movePoint(Point mapPoint) {
+
+    // set the selected features' geometry to a new map point
+    selectedFeature.setGeometry(mapPoint);
+
+    // update the selected feature's feature table
+    updateAttributes(selectedFeature);
+
+    // clear selection of the point
+    clearSelection();
+    selectedFeature = null;
+  }
+
+  /**
+   * Move the last of the vertex point of the currently selected polyline to the given map point by updating the
+   * selected feature's geometry and feature table.
+   */
+  private void movePolylineVertex(Point mapPoint) {
+
+    // get the selected feature's geometry as a polyline
+    Polyline polyline = (Polyline) selectedFeature.getGeometry();
+
+    // create a polyline builder to add and remove parts from the polyline
+    PolylineBuilder polylineBuilder = new PolylineBuilder(polyline);
+
+    // get the nearest vertex to the map point on the polyline
+    ProximityResult nearestVertex = GeometryEngine.nearestVertex(polyline, (Point) GeometryEngine.project(mapPoint, polyline.getSpatialReference()));
```

Comment:
This is a really long line, so let's just break it down for readability:
```suggestion
    ProximityResult nearestVertex =
            GeometryEngine.nearestVertex(polyline, (Point) GeometryEngine.project(mapPoint, polyline.getSpatialReference()));
```